src/scrapers/otsar-hachayal.js

```diff
@@ -0,0 +1,222 @@
+  // TODO: support change password
+  /* urls[LOGIN_RESULT.CHANGE_PASSWORD] = [``]; */
+  return urls;
+}
+
+function getTransactionsUrl() {
+  return `${BASE_URL}/wps/myportal/FibiMenu/Online/OnAccountMngment/OnBalanceTrans/PrivateAccountFlow`;
+}
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#username', value: credentials.userCode },
+    { selector: '#password', value: credentials.password },
+  ];
+}
+
+function getAmountData(amountStr, hasCurrency = false) {
+  const amountStrCopy = amountStr.replace(',', '');
+  let currency = null;
+  let amount = null;
+  if (!hasCurrency) {
+    const amountStrCopy = amountStr.replace(',', '');
+    amount = parseFloat(amountStrCopy);
+    currency = SHEKEL_CURRENCY;
+  } else if (amountStrCopy.includes(SHEKEL_CURRENCY_SYMBOL)) {
+    amount = parseFloat(amountStrCopy.replace(SHEKEL_CURRENCY_SYMBOL, ''));
+    currency = SHEKEL_CURRENCY;
+  } else {
+    const parts = amountStrCopy.split(' ');
+    amount = parseFloat(parts[0]);
+    [, currency] = parts;
+  }
+
+  return {
+    amount,
+    currency,
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((txn) => {
+    const txnDate = moment(txn.date, DATE_FORMAT).toISOString();
+
+    const credit = getAmountData(txn.credit).amount;
+    const debit = getAmountData(txn.debit).amount;
+    const amount = (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+      date: txnDate,
+      processedDate: txnDate,
+      originalAmount: amount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: amount,
+      description: txn.description,
+    };
+  });
+}
+
+async function readPage(page) {
+  const tdsValues = await page.$$eval('#dataTable077 tbody tr td', (tds) => {
+    return tds.map(td =>
+      ({
+        classList: td.getAttribute('class'),
+        innerText: td.innerText,
+      }));
+  });
+
+  const txns = [];
+  for (const element of tdsValues) {
+    const { classList, innerText } = element;
+    if (classList.includes('date')) {
+      const newTransaction = {};
+      newTransaction.date = innerText;
+      txns.push(newTransaction);
+    } else {
+      const changedTransaction = txns.pop();
+      if (classList.includes('reference')) {
+        changedTransaction.description = innerText;
+      } else if (classList.includes('details')) {
+        changedTransaction.reference = innerText;
+      } else if (classList.includes('credit')) {
+        changedTransaction.credit = innerText;
+      } else if (classList.includes('debit')) {
+        changedTransaction.debit = innerText;
+      } else if (classList.includes('balance')) {
+        changedTransaction.balance = innerText;
+      }
+      txns.push(changedTransaction);
+    }
+  }
+
+  return txns;
+}
+
+async function getAccountSummary(page) {
+  let summary;
+  try {
```

Comment:
@matanelgabsi can you get rid of the `try..catch` clause here?
If we fail anywhere, it should be handled by the base class, and not return an empty object.